notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/main/kotlin/com/r3/corda/notary/plugin/nonvalidating/server/NonValidatingNotaryServerFlowImpl.kt

```diff
@@ -111,6 +122,21 @@ class NonValidatingNotaryServerFlowImpl() : ResponderFlow {
         }
     }
 
+    /**
+     * This function will validate selected notary is valid notary to notarize.
+     * */
+    @Suspendable
+    private fun getCurrentNotaryAndValidateNotary(txDetails: NonValidatingNotaryTransactionDetails) {
+        val currentNotaryMemberProvidedCtx = memberLookup.myInfo().memberProvidedContext
+        val currentNotaryServiceName = currentNotaryMemberProvidedCtx.parse(NOTARY_SERVICE_NAME, MemberX500Name::class.java)
+
+        val payloadNotaryServiceName = txDetails.notaryName
+
+        require(currentNotaryServiceName == payloadNotaryServiceName) {
+            "Given notary service = $payloadNotaryServiceName is invalid"
```

Comment:
```suggestion
            "Notary service on the transaction $payloadNotaryServiceName does not match the notary service represented by this notary virtual node (${currentNotaryServiceName})"
```

Current message is a bit misleading (it may well be a valid notary service). This text is more accurate and provides a more specific / helpful error.